lib/cloudant.js

```diff
@@ -522,6 +524,19 @@ Cloudant.prototype.updateAttributes = function(model, id, data, options, cb) {
 };
 
 /**
+ * If input is a model instance, convert to a plain JSON object
+ * The input would be a model instance when the request is made from
+ * REST endpoint as remoting converts it to model if endpoint expects a
+ * model instance
+ */
+
+function getPlainJSONData(model, data) {
+  if (this._models[model] && data instanceof this._models[model].model) {
+    return data.toJSON();
+  }
+  return data;
+}
```

Comment:
It might be that I don't understand exactly what's happening here, but it seems to me like this function is the sort of thing that would happen in the base connector repository, and not in this particular one. How is this handled in other connectors?